src/Dialog/withResponsiveFullScreen.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+
+import createEagerFactory from 'recompose/createEagerFactory';
+import wrapDisplayName from 'recompose/wrapDisplayName';
+import withWidth, { isWidthDown } from '../utils/withWidth';
+import Dialog from './Dialog';
+import type { Breakpoint } from '../styles/breakpoints';
+
+type Options = { breakpoint: Breakpoint }
+
+/**
+ * Dialog will responsively be full screen _at or below_ the given breakpoint
+ * (defaults to 'sm' for mobile devices).
+ */
+function withResponsiveFullScreen(options: Options = { breakpoint: 'sm' }) {
```

Comment:
Yeah, I like this Higher-order Component ❤️ . We could imagine making it even more generic with a property modifier function. Some idead I have in mind. I'm not saying it should be like this.
```js
import Dialog from 'material-ui/Dialog'
import withResponsiveProps from 'material-ui/utils/withResponsiveProps'

const ResponsiveDialog = withResponsiveProps({
  breakpoints: 'md',
  switch: (props, match) => ({
    fullscrean: match ? true : false,
    ...props,
  })
})(Dialog)

```

or

```js
import Dialog from 'material-ui/Dialog'
import ResponsiveProps from 'material-ui/utils/withResponsiveProps'

<ResponsiveProps
  breakpoints="md"
  switch={(props, match) => ({
    fullscrean: match ? true : false,
  })}
>
  <Dialog />
</ResponsiveProps>


```